### DISCUSSION TITLE
Learner's question - how do I get the CLI to survive a reboot on Linux (maybe a systemd service file)?

### DISCUSSION THREAD

**Discussion by bouton** [maplibre/martin#1586] [Q&A]
Sorry - new to Linux and trying this out.

I have Martin up and running displaying all my tiles with maplibre. Lovely.

To keep it running in the background  I run it with an ampersand at the end of the command
eg martin /path/etc &

But a reboot requires me to re-enter the command. 

I have read up that maybe I can run it as a systemd service.  Is this possible?

Do you have an example service file and/or some suggestions? Thanks

**Comment by CommanderStorm** [comment]
Deploying this as a systemd service is not quite the simplest solution imo (might be biased).
I would suggest looking into [docker](https://maplibre.org/martin/run-with-docker.html) instead. Much simpler than [the systemd docs](https://www.freedesktop.org/software/systemd/man/latest/systemd.service.html).

If you still want to go via the systemd route, I would need a few additional pieces of information:
- what are your datasources (affects what startup is blocked by)
- if using the `--auto-bounds` parameter: what is the datasources size and is extent indexed (affects startup time)
- when do you need it to start up (affects what startup is blocked by; immediately after networking, or is your system constrained on resources and another service start first. martins resource usage is low => should not affect you much if at all)

**Comment by bouton** [comment]
Thanks.  Will look into docker per your suggestion. Thanks

**Reply by sharkAndshark** [reply]
I remember [a basic systemd support](https://github.com/maplibre/martin/pull/761) is added  but not documented in the Martin book. If you are using Debian/Ubuntu you could get system d support by installing Martin with Martin.deb which can be downloaded in the release page.
